bats/tests/k8s/helm-install-rancher.bats

```diff
@@ -13,12 +13,95 @@ local_setup() {
     helm repo update
 }
 
+# Check that the rancher-latest/rancher helm chart at the given version is
+# supported on the current Kubernetes version (as determined by
+# $RD_KUBERNETES_PREV_VERSION)
+assert_rancher_chart_compatible() {
+    output="" # Hack: tell shellcheck about this variable
+    local chart_version=$1
+
+    run helm show chart rancher-latest/rancher --version "$chart_version"
+    assert_success || return
+
+    run awk '/^kubeVersion:/ { $1 = ""; print }' <<<"$output"
+    assert_success || return
+    # We only support kubeVersion of form "< x.y.z"
+    assert_output --regexp '^[[:space:]]*<[[:space:]]*[^[:space:]]+$' || return
+
+    run awk '{ print $2 }' <<<"$output"
+    assert_success || return
+
+    local unsupported_version=$output
+    semver_gt "$unsupported_version" "$RD_KUBERNETES_PREV_VERSION" || return
+}
+
+# Set (and save) $rancher_chart_version to $RD_RANCHER_IMAGE_TAG if it is set
+# (and compatible), or otherwise the oldest chart version that supports
+# $RD_KUBERNETES_PREV_VERSION.
+# If no compatible chart version could be found, calls mark_k3s_version_skipped
+# and fails the test.
+determine_chart_version() {
+    local rancher_chart_version
+    if [[ -n $RD_RANCHER_IMAGE_TAG ]]; then
+        # If a version is given, check that it's compatible.
+        rancher_chart_version=${RD_RANCHER_IMAGE_TAG#v}
+        if ! assert_rancher_chart_compatible "$rancher_chart_version"; then
```

Comment:
I continue to think `assert_*` is the wrong name if you use it as a predicate function instead of `assert_* || return`. Otherwise you could prefix any function that returns a boolean with `assert_`.